search-cache-pipeline.yml

```diff
@@ -128,7 +128,7 @@ extends:
           displayName: Install latest daily .NET version
 
         - bash: >
-            $(Build.SourcesDirectory)/.dotnet/dotnet $(Build.SourcesDirectory)/artifacts/bin/Microsoft.TemplateSearch.TemplateDiscovery/Debug/net10.0/Microsoft.TemplateSearch.TemplateDiscovery.dll --basePath $(System.DefaultWorkingDirectory)/NugetDownloadDirectory --allowPreviewPacks -v --test --diff $(EnableDiffMode)
+            $(Build.SourcesDirectory)/.dotnet/dotnet $(Build.SourcesDirectory)/artifacts/bin/Microsoft.TemplateSearch.TemplateDiscovery/Debug/net8.0/Microsoft.TemplateSearch.TemplateDiscovery.dll --basePath $(System.DefaultWorkingDirectory)/NugetDownloadDirectory --allowPreviewPacks -v --test --diff $(EnableDiffMode)
```

Comment:
I see a few low cost solutions to avoid the hard coded TFM:

- Use `dotnet run` to build and run the application without supplying a TFM value (my preferred solution)
- Set `AppendTargetFrameworkToOutputPath` false in the project. Feels hacky to me.